### PR TITLE
Authorization cancancan

### PR DESCRIPTION
**Pull request by Favourezeugwa**
### I have Implemented the following in this pull request
- Installed cancancan gem
- Add conditions in Ability.rb file
- Modify nav bar(removed shopping cart)
- Add authorize_user to controllers where necessary

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    can :read, Recipe if recipe.public?
+
+    return unless user.present?
+
+    can :manage, :all, user_id: user.id
+
+    return unless user.role == 'admin'
+
+    can :manage, :all
+    #   return unless user.present?
```
Comment (assadounto) [inline app/models/ability.rb]: Please I suggest that we refactor the roles for  a regular user  and that of an admin
into something like this

```suggestion
     user ||= User.new # guest user (not logged in)
    if user.role == 'admin'
      can :manage, :all
    else
      can :read, Recipe do |recipe|
        recipe.user == user || recipe.public
      end

      can %i[update destroy], Recipe do |recipe|
        recipe.user == user
      end

      can :read, Food
      can :destroy, Food do |food|
        food.user.id == user.id
      end
      can %i[create], :all
    end
```

**Review by assadounto — CHANGES_REQUESTED** [review]
Great job @Favourezeugwa  implementing the required feature! ✅✅✅
However there are some  little changes you need to make to make this feature ready

Required Changes ♻
1. Kindly implement the authorization on the view so that only users allowed to access certain parts and perform actions are done correctly as stated in the project requirements . I am thinking you missed that part.

2. Please  check inline comments

File: app/models/ability.rb (hunk 1 above)
Comment (Favourezeugwa) [reply app/models/ability.rb]: Thank you @assadounto for your kind suggestions, I have implemented the required changes 😄

**Review by assadounto — APPROVED** [review]
Congrats @Favourezeugwaon implementing all requirements🤗



There is not much to say than its merge time!  🎈🎈

**Pull request merged**